src/test/java/de/numcodex/feasibility_gui_backend/query/broker/direct/DirectBrokerClientCqlTest.java

```diff
@@ -0,0 +1,87 @@
+package de.numcodex.feasibility_gui_backend.query.broker.direct;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.rest.client.api.IGenericClient;
+import de.numcodex.feasibility_gui_backend.query.broker.QueryNotFoundException;
+import de.numcodex.feasibility_gui_backend.query.broker.SiteNotFoundException;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.web.reactive.function.client.WebClient;
+
+@ExtendWith(MockitoExtension.class)
+class DirectBrokerClientCqlTest {
+
+    private static final Long TEST_BACKEND_QUERY_ID = 1L;
+
+    @SuppressWarnings("unused")
+    @Mock
+    FhirContext fhirContext;
+
+    @Mock
+    IGenericClient fhirClient;
+
+    @InjectMocks
+    DirectBrokerClientCql client;
+
+    @Test
+    void testPublishNonExistingQuery() {
+        assertThrows(QueryNotFoundException.class, () -> client.publishQuery("does-not-exist"));
+    }
+
+    @Test
+    void testPublishExistingQueryWithoutStructuredQueryDefinition() {
+        var queryId = client.createQuery(TEST_BACKEND_QUERY_ID);
+        assertThrows(IllegalStateException.class, () -> client.publishQuery(queryId));
+    }
+
```

Comment:
Please add tests for IOException on publish.

Reply:
I'm not quite sure how to do that. The exception will be thrown when a method fails to load a resource file. All calls there are either private or static, so I can't mock them using a Spy. I could mock the publishQuery method itself, but that would render the whole test kinda moot.

Reply:
Still struggling with this one...gotta get back to it tomorrow